File: conformance/bundles/12-combining-marks/greet.steps.rs
//! Rust sibling of `greet.steps.ts` (bundle `12-combining-marks`).

use var::{Handler, Registry, Steps, Value};

pub const FILE: &str = "greet.steps.rs";

pub fn register(r: Registry) -> Registry {
    let mut s = Steps::from_registry(r);
    s.sensor(
        "I greet {string}",
        FILE,
        1,
        Handler::sync1(|_state, _name| Ok(None)),
    );
    s.into_registry()
}

pub fn state() -> Value {
    Value::Null
}

File: conformance/bundles/12-combining-marks/greet.steps.cs
// C# sibling of greet.steps.ts / .rs (bundle 12-combining-marks).
using Varar;
using Varar.Core;

namespace Varar.Corpus.B12;

public static class GreetSteps
{
    public static Registry Register(Registry r)
    {
        var s = Steps.From(r);

        s.Sensor("I greet {string}", (state, name) => null);

        return s.ToRegistry();
    }

    public static Value State() => Value.Null;
}
